WordPress/src/main/java/org/wordpress/android/ui/uploads/UploadService.java

```diff
@@ -107,9 +112,14 @@ public void onDestroy() {
             mPostUploadHandler.unregister();
         }
 
-        // Update posts with any completed uploads in our post->media map
-        for (Integer postId : sCompletedMediaByPost.keySet()) {
+        // Update posts with any completed AND failed uploads in our post->media map
+        ConcurrentHashMap<Integer, List<MediaModel>> tmpMap = new ConcurrentHashMap<>();
```

Comment:
Looking at this again, we're not actually using the `List<MediaModel>` part of the map - we only want a list of the post ids with either completed or failed media. So what if we just did something like:

```java
List<Integer> postIdList = new ArrayList<>();
postIdList.addAll(sCompletedMediaByPost.keySet());
postIdList.addAll(sFailedMediaByPost.keySet());
```

and dropped the `ConcurrentHashMap<Integer, List<MediaModel>> tmpMap`?